File: Shubham Garad/Microsoft/EvaluateReversePolishNotation.cpp
#define ll long long
class Solution
{
    public:
        ll cal(ll num1 ,ll num2, char &op)
        {
            switch (op)
            {

                case '+':
                    return num1 + num2;
                    break;

                case '-':
                    return num1 - num2;
                    break;

                case '*':
                    return num1 * num2;
                    break;

                case '/':
                    return num1 / num2;
                    break;
            }
            return 0;
        }
    ll evalRPN(vector<string> &tokens)
    {
        stack<ll> s;
        for (auto &i: tokens)
        {
            if (i == "-"
                or i == "+"
                or i == "*"
                or i == "/")
            {
                ll a = s.top();
                s.pop();
                ll b = s.top();
                s.pop();
                s.push(cal(b, a, i[0]));
            }
            else
            {
                s.push(stoi(i));
            }
        }
        return s.top();
    }
};

File: RAO.sh/Microsoft/Day I/EvaluateReversePolishNotation.go
func evalRPN(tokens []string) int {
	var stack []int

	for _, token := range tokens {
		if token != "+" && token != "-" && token != "*" && token != "/" {
			num, _ := strconv.Atoi(token)
			stack = append(stack, num)
		} else {
			num1 := stack[len(stack)-1]
			stack = stack[:len(stack)-1]
			num2 := stack[len(stack)-1]
			stack = stack[:len(stack)-1]

			switch token {
			case "+":
				stack = append(stack, num2+num1)
			case "-":
				stack = append(stack, num2-num1)
			case "*":
				stack = append(stack, num2*num1)
			case "/":
				stack = append(stack, num2/num1)

			}
		}

	}

	return stack[0]
}
